pkg/injector/patcher/sidecar.go

```diff
@@ -25,6 +25,12 @@ import (
 	"github.com/dapr/dapr/utils"
 )
 
+const (
+	defaultSoftLimitPercentage    = 80
```

Comment:
Shouldn't the default just be no soft limit?

Reply:
it could be, just wanted to have some best practice to avoid OOM kill. I've increased it to 90% to follow recs in [golang gc doc](https://tip.golang.org/doc/gc-guide#Memory_limit).  This is only applied if there is a memory limit set by the user if there is no limit assigned by the user there is no percentage limit.